examples/pong/src/pong.rs

```diff
@@ -0,0 +1,400 @@
+
+            if !self.p2_racket.is_colliding_y(predicted_pos) {
+                if predicted_pos as i32 > half_screen {
+                    PongGame::update_racket_pos(&mut self.p2_racket, RACKET_MOVEMENT_SPEED)
+                } else {
+                    PongGame::update_racket_pos(&mut self.p2_racket, -RACKET_MOVEMENT_SPEED)
+                };
+            }
+        }
+    }
+
+    fn update_p1_racket_pos(&mut self, gamepad: u8) {
+        if gamepad & BUTTON_UP != 0 {
+            PongGame::update_racket_pos(&mut self.p1_racket, -RACKET_MOVEMENT_SPEED);
+        } else if gamepad & BUTTON_DOWN != 0 {
+            PongGame::update_racket_pos(&mut self.p1_racket, RACKET_MOVEMENT_SPEED)
+        }
+    }
+
+    #[inline]
+    fn update_ball_pos(&mut self) -> CollisionType {
+        self.ball.rect.move_by(self.ball.vx, self.ball.vy);
+        let mut collision = CollisionType::empty();
+
+        if self.ball.rect.is_colliding_rect(&self.p1_racket) {
+            collision |= CollisionType::P1_RACKET;
+            self.ball.vx = -self.ball.vx;
+        } else if self.ball.rect.is_colliding_rect(&self.p2_racket) {
+            collision |= CollisionType::P2_RACKET;
+            self.ball.vx = -self.ball.vx;
+        }
+
+        if self.ball.rect.is_colliding_rect(&GAME_FRAME.top) {
+            collision |= CollisionType::TOP;
+            self.ball.vy = -self.ball.vy;
+        } else if self.ball.rect.is_colliding_rect(&GAME_FRAME.bottom) {
+            collision |= CollisionType::BOTTOM;
+            self.ball.vy = -self.ball.vy;
+        }
+
+        if self.ball.rect.is_colliding_rect(&GAME_FRAME.right) {
+            collision |= CollisionType::RIGHT;
+        } else if self.ball.rect.is_colliding_rect(&GAME_FRAME.left) {
+            collision = CollisionType::LEFT;
+        }
+
+        if !collision.is_empty() {
+            self.ball.rect.move_by(self.ball.vx, self.ball.vy);
+        }
+
+        collision
+    }
+
+    fn play_tone(&mut self, frequency: u32, volume: u32, duration: u32, flags: u32) {
+        if self.disable_tone_frames_left == 0 {
+            self.disable_tone_frames_left = MIN_FRAMES_BETWEEN_TONES;
+            tone(frequency, volume, duration, flags);
+        }
+    }
+
+    fn update_palette(&mut self) {
+        if self.disable_palette_change_frame_left == 0 {
+            self.disable_palette_change_frame_left = MIN_FRAMES_BETWEEN_PALETTE_CHANGE;
+            self.palette = self.palette.overflowing_add(1).0;
+            change_palette(self.palette);
+        }
+    }
+
+    fn tick(&mut self, gamepad: u8) {
+        self.update_p1_racket_pos(gamepad);
+        self.update_p2_racket_pos();
+        let collision = self.update_ball_pos();
+
+        if collision.intersects(CollisionType::P1_RACKET) {
+            self.play_tone(240, 3, 1, TONE_TRIANGLE);
+        } else if collision.intersects(CollisionType::P2_RACKET) {
+            self.play_tone(262, 3, 1, TONE_TRIANGLE);
```

Comment:
Should the volume here be higher? It's almost muted at 1%

Reply:
It sounded ok during dev, but you're right it too low.

I'll submit a PR.